### PR TITLE
[stdlib] Sendablize all the things

### PR DESCRIPTION
**Pull request by Azoy**
This PR adds sendable information to all public/UFI types to produce 0 warnings with `-require-explicit-sendable`. This patch doesn't turn that flag on yet because we'll have to audit other standard library modules to conform.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

File: stdlib/public/core/DropWhile.swift [hunk 1]
```diff
@@ -28,9 +28,11 @@ public struct LazyDropWhileSequence<Base: Sequence> {
     self._base = _base
     self._predicate = predicate
   }
-
 }
 
+@available(*, unavailable)
+extension LazyDropWhileSequence: Sendable {}
```
Comment (FranzBusch) [inline stdlib/public/core/DropWhile.swift:34]: Why isn't this type conditionally Sendable?

File: stdlib/public/core/Filter.swift [hunk 2]
```diff
@@ -36,6 +36,9 @@ public struct LazyFilterSequence<Base: Sequence> {
   }
 }
 
+@available(*, unavailable)
+extension LazyFilterSequence: Sendable {}
```
Comment (FranzBusch) [inline stdlib/public/core/Filter.swift:40]: Same here

File: stdlib/public/core/DropWhile.swift (hunk 1 above)
Comment (Azoy) [reply stdlib/public/core/DropWhile.swift:34]: The predicate it stores is not known to be sendable. Same with `LazyFilter`

Comment (FranzBusch) [reply stdlib/public/core/DropWhile.swift:34]: Ah that's interesting. We had the same problem with some of async variants and they all require the closure to be Sendable but I agree that doesn't make sense here!

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Pull request merged**